driver-core/src/main/java/com/datastax/driver/core/DateWithoutTime.java

```diff
@@ -0,0 +1,260 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * A date with no time components, no time zone, in the ISO 8601 calendar.
+ *
+ * Note that ISO 8601 has a number of differences with the default gregorian calendar used in Java:
+ * <ul>
+ *     <li>it uses a proleptic gregorian calendar, meaning that it's gregorian indefinitely back in the past (there is no gregorian change);</li>
+ *     <li>there is a year 0.</li>
+ * </ul>
+ *
+ * This class implements these differences, so that year/month/day fields match exactly the ones in
+ * CQL string literals.
+ *
+ * @since 2.2
+ */
+public final class DateWithoutTime {
+
+    private static final TimeZone UTC = TimeZone.getTimeZone("UTC");
+
+    private final long millisSinceEpoch;
+    private final int daysSinceEpoch;
+
+    // This gets initialized lazily if we ever need it. Once set, it is effectively immutable.
+    private volatile GregorianCalendar calendar;
```

Comment:
Ah, I got the getCalendar and isoCalendar() implementations mixed up, makes sense that you want a unique calendar per instance.